assignment5/assignment5.cpp

```diff
@@ -1,60 +1,103 @@
 #include <iostream>
 #include <vector>
+#include <map>
 
-class UnknownLanguage
-{
-	std::vector<std::string> dictionary;
+class UnknownLanguage{
+	struct Node{
+		std::string in_edges;
+		std::string out_edges;	
+	};
+	std::map<char, Node> graph;
 	std::string alphabet;
-public:
-	UnknownLanguage(std::vector<std::string> dic):dictionary(dic){}
-	std::string CalculateAlphabet(){
+	std::vector<std::string> dict;
+	void DictionaryToGraph(){
+		graph.clear();
+		for(int i=0; i<dict.size()-1; i++){
```

Comment:
Very subtle bug here: dict.size() is unsigned, if it's 0, subtracting 1 gives you a very big number, things don't go well after that...